libgnucash/engine/Account.cpp

```diff
@@ -3988,6 +3988,50 @@ xaccAccountGetNoclosingBalanceChangeForPeriod (Account *acc, time64 t1,
     return gnc_numeric_sub(b2, b1, GNC_DENOM_AUTO, GNC_HOW_DENOM_FIXED);
 }
 
+typedef struct
+{
+    const gnc_commodity *currency;
+    gnc_numeric balanceChange;
+    time64 t1;
+    time64 t2;
+} CurrencyBalanceChange;
+
+static void
+xaccAccountBalanceChangeHelper (Account *acc, gpointer data)
+{
+    CurrencyBalanceChange *cbdiff = static_cast<CurrencyBalanceChange*>(data);
+
+    gnc_numeric b1, b2;
+    b1 = GetBalanceAsOfDate(acc, cbdiff->t1, TRUE);
+    b2 = GetBalanceAsOfDate(acc, cbdiff->t2, TRUE);
+    gnc_numeric balanceChange = gnc_numeric_sub(b2, b1, GNC_DENOM_AUTO, GNC_HOW_DENOM_FIXED);
+    gnc_numeric balanceChange_conv = xaccAccountConvertBalanceToCurrencyAsOfDate(acc, balanceChange,xaccAccountGetCommodity(acc),cbdiff->currency,cbdiff->t1);
```

Comment:
Don't you want the ending rate, i.e. `cbdiff->t2`?

Also spaces after commas please.